core/src/main/java/org/apache/iceberg/encryption/StandardEncryptionManager.java

```diff
@@ -20,28 +20,46 @@
 
 import java.nio.ByteBuffer;
 import java.security.SecureRandom;
+import java.util.Base64;
+import java.util.Map;
+import org.apache.iceberg.CatalogProperties;
 import org.apache.iceberg.TableProperties;
 import org.apache.iceberg.io.InputFile;
 import org.apache.iceberg.io.OutputFile;
 import org.apache.iceberg.io.SeekableInputStream;
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
 import org.apache.iceberg.relocated.com.google.common.collect.Iterables;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
 import org.apache.iceberg.util.ByteBuffers;
 
 public class StandardEncryptionManager implements EncryptionManager {
+  public static final int KEK_ID_LENGTH = 16;
+
   private final transient KeyManagementClient kmsClient;
   private final String tableKeyId;
   private final int dataKeyLength;
-
+  private final long kekCacheTimeout;
+  private Map<String, KeyEncryptionKey> kekCache;
   private transient volatile SecureRandom lazyRNG = null;
 
+  /**
+   * @deprecated will be removed in 2.0.0. use {@link #StandardEncryptionManager(String, int,
+   *     KeyManagementClient, long)} instead.
+   */
+  @Deprecated
+  public StandardEncryptionManager(
+      String tableKeyId, int dataKeyLength, KeyManagementClient kmsClient) {
+    this(tableKeyId, dataKeyLength, kmsClient, CatalogProperties.KEK_CACHE_TIMEOUT_MS_DEFAULT);
+  }
+
   /**
    * @param tableKeyId table encryption key id
    * @param dataKeyLength length of data encryption key (16/24/32 bytes)
    * @param kmsClient Client of KMS used to wrap/unwrap keys in envelope encryption
+   * @param kekCacheTimeout timeout of kek (key encryption key) cache entries
    */
   public StandardEncryptionManager(
-      String tableKeyId, int dataKeyLength, KeyManagementClient kmsClient) {
+      String tableKeyId, int dataKeyLength, KeyManagementClient kmsClient, long kekCacheTimeout) {
```

Comment:
Do we need a new constructor? Seems like we're just going to use a default value and almost never change it.

Reply:
Different organizations can set the kekCacheTimeout differently, to address their particular security requirements. But that's probably an advanced feature. We can start with a default value; and make it configurable later - if needed (if we get a request from the community). What do you think?

Reply:
If Flink produces a new snapshot every 2 minutes, we will have by default 144 KEKs a day.. Growing to a very large number in days and weeks. All of them unwrapped by KMS in readers, can be very expensive. We might want to keep this parameter configurable..
(now it is renamed to "writer KEK timeout")

Reply:
Actually, we have an explicit NIST recommendation for this in  https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-57pt1r5.pdf
_"The recommended originator-usage period for a symmetric key-wrapping
key that is used to wrap very large numbers of keys over a short period of time is on the
order of a day or a week. If a relatively small number of keys are to be wrapped under a
key-wrapping key, the originator-usage period of the key-wrapping key could be up to
two years. "_